sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/JobDisablePollerAsync.java

```diff
@@ -0,0 +1,113 @@
+ * Returns {@link BatchJob} values during polling and as the final result.
+ */
+public final class JobDisablePollerAsync {
+
+    private final BatchAsyncClient batchAsyncClient;
+    private final String jobId;
+    private final BatchJobDisableParameters parameters;
+    private final RequestOptions options;
+    private final Context requestContext;
+
+    /**
+     * Creates a new {@link JobDisablePollerAsync}.
+     *
+     * @param batchAsyncClient The {@link BatchAsyncClient} used to interact with the Batch service.
+     * @param jobId The ID of the Job being disabled.
+     * @param parameters The options to use when disabling the Job.
+     * @param options Optional request options for service calls.
+     */
+    public JobDisablePollerAsync(BatchAsyncClient batchAsyncClient, String jobId, BatchJobDisableParameters parameters,
+        RequestOptions options) {
+        this.batchAsyncClient = batchAsyncClient;
+        this.jobId = jobId;
+        this.parameters = parameters;
+        this.options = options;
+        this.requestContext = options == null ? Context.NONE : options.getContext();
+    }
+
+    /**
+     * Activation operation to start the disable.
+     *
+     * @return A function that initiates the disable request and returns a PollResponse with IN_PROGRESS status.
+     */
+    public Function<PollingContext<BatchJob>, Mono<PollResponse<BatchJob>>> getActivationOperation() {
+        return context -> batchAsyncClient
+            .disableJobWithResponse(jobId, com.azure.core.util.BinaryData.fromObject(parameters), options)
+            .then(batchAsyncClient.getJobWithResponse(jobId, null))
+            .map(response -> {
+                BatchJob job = response.getValue().toObject(BatchJob.class);
+                return new PollResponse<>(LongRunningOperationStatus.IN_PROGRESS, job);
+            });
+    }
+
+    /**
+     * Poll operation to check the job's disablement state.
+     *
+     * @return A function that polls the Job and returns a PollResponse with the current operation status.
+     */
+    public Function<PollingContext<BatchJob>, Mono<PollResponse<BatchJob>>> getPollOperation() {
+        return context -> {
+            RequestOptions pollOptions = new RequestOptions().setContext(this.requestContext);
+            return batchAsyncClient.getJobWithResponse(jobId, pollOptions).flatMap(response -> {
+                BatchJob job = response.getValue().toObject(BatchJob.class);
+                BatchJobState state = job.getState();
+
+                LongRunningOperationStatus status;
+                if (BatchJobState.DISABLING.equals(state)) {
+                    status = LongRunningOperationStatus.IN_PROGRESS;
+                } else if (BatchJobState.DISABLED.equals(state)) {
+                    status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+                } else {
+                    status = LongRunningOperationStatus.FAILED;
```

Comment:
We shouldn't treat this as a failure. Who's to say the job didn't get re-enabled between polling cycles? I suspect we have this bug in a bunch of the pollers.